src/rdma_transport.h

```diff
@@ -412,14 +416,13 @@ class RDMATransport : public Transport {
   virtual int RecvPushRequest(Message *msg, BufferContext *buffer_ctx, int meta_len) {
     CHECK(msg->meta.push && msg->meta.request);
     CHECK_EQ(buffer_ctx->data_num, 3);
-    uint32_t len = buffer_ctx->data_len[1];
     char* cur = buffer_ctx->buffer + align_ceil((size_t) meta_len, pagesize_);
-    
+
     SArray<char> keys = CreateFunctionalSarray(&msg->meta.key, sizeof(Key));
 
+    uint32_t len = msg->meta.val_len;
```

Comment:
a related question: does any code rely on the assumption that `msg->meta.val_len == 0` for pull requests? it's still 0 in the ucx van.

Reply:
No. Actually some code assumes it won't be 0. Are you using the latest ucx van? the value of val_len is preserved after #57

Reply:
ok I see what's going on. The squash/merge of pr-56 reverted it back: https://github.com/bytedance/ps-lite/commit/0bd64f2782e51c392c81563f6e99e72bdaea188e#diff-1271bc61af3c78df47fe583c8841d84d55a39a990a814ae4de73812da0e16d58R856